sphinxcontrib/needs/needs.py

```diff
@@ -514,6 +518,21 @@ def check_configuration(_app: Sphinx, config: Config):
     extra_options = config["needs_extra_options"]
     link_types = [x["option"] for x in config["needs_extra_links"]]
 
+    external_filter = config["needs_filter_data"]
```

Comment:
what if "needs_filter_data" does not exist?
use get() to get values from dicts safely

Reply:
It's fine here, config["needs_filter_data"] is a dictionary.

Reply:
yes I know, but ``config.get('needs_filter_data', {})`` is more save, as it always set a value.
In your case ``config["needs_filter_data"]`` can be None, then ``external_filter`` is None and ``eval()`` will raise an exception.

I know, Sphinx should have set ``{}`` as default value for ``needs_filter_data``, but i would not trust this.